### ISSUE TITLE
Should a non-symbol Expr be callable?

### ISSUE THREAD

**Issue by darius** [aimacode/aima-python#216]
A symbol like `f` can be called like `f(0)`, using this code: https://github.com/aimacode/aima-python/blob/master/utils.py#L397

But that code allows weird uses like `(a+b)(1)` meaning `Expr('+', 1)`. Probably this should raise an error instead when `self.args` is nonempty, as in this case.

(I'd go ahead and make this change without discussion except that I'm not set up to run any of the tests in iPython notebooks.)

**Comment by norvig** [comment]
I agree. There might be some complex math where you want to do that, but mostly you don't want to. And in particular, if someone translates the math expression `(x + 1) (x - 1)` verbatim into Python, rather than into `(x + 1) * (x - 1)`, then we'd rather have it give an error than to be a function call. I'll fix it.

**Comment by dmeoli** [comment]
Already addressed on current master. `Expr.__call__` now guards against calling a non-Symbol Expr (utils.py): `if self.args: raise ValueError('Can only do a call for a Symbol, not an Expr')`. So `P(x)(y)` and `(a + b)(1)` both raise `ValueError` instead of silently producing a nonsensical Expr. Closing as resolved.
